### PR TITLE
handle forbidden error when checking new namespace

### PR DESCRIPTION
**Pull request by gitlawr**
Address Issue https://github.com/rancher/rancher/issues/16132

### CODE REVIEW

File: cmd/app.go [hunk 1]
```diff
@@ -5,7 +5,9 @@ import (
 	"encoding/base64"
 	"encoding/json"
 	"fmt"
+	"github.com/rancher/norman/clientbase"
```
Comment (orangedeng) [inline cmd/app.go]: This might have a go fmt problem

**Comment by orangedeng** [comment]
LGTM

File: cmd/app.go [hunk 2]
```diff
@@ -787,14 +789,18 @@ func createNamespace(c *cliclient.MasterClient, n string) error {
...
 		startTime := time.Now()
 		for {
 			logrus.Debug(fmt.Sprintf("Namespace create wait - Name: %s, State: %s, Transitioning: %s", ns.Name, ns.State, ns.Transitioning))
 			if time.Since(startTime)/time.Second > 30 {
 				return fmt.Errorf("timed out waiting for new namespace %s", ns.Name)
 			}
-			ns, err = c.ClusterClient.Namespace.ByID(ns.ID)
-			if err != nil {
+			ns, err = c.ClusterClient.Namespace.ByID(nsID)
+			if e, ok := err.(*clientbase.APIError); ok && e.StatusCode == http.StatusForbidden {
```
Comment (dramich) [inline cmd/app.go]: This check should be inside the `if err != nil {` if we don't have an error we don't need to type assert the error.
```
			ns, err = c.ClusterClient.Namespace.ByID(nsID)
			if err != nil {
				if e, ok := err.(*clientbase.APIError); ok && e.StatusCode == http.StatusForbidden {
					continue
				}
				return err
			}
```

Comment (gitlawr) [reply cmd/app.go]: make sense

**Comment by dramich** [comment]
LGTM

**Pull request merged**